api/v1beta1/disruption_webhook.go

```diff
@@ -316,7 +329,40 @@ You first need to remove those chaos pods (and potentially their finalizers) to
 	return nil
 }
 
-func (r *Disruption) validateUserInfo(oldDisruption *Disruption) error {
+// validateUserInfoGroup checks that if permittedUserGroups is set, which is controlled in controller.safeMode.permittedUserGroups in the configmap,
+// then we will return an error if the user in r.UserInfo does not belong to any groups. If permittedUserGroups is unset, or if the user belongs to one of those
+// groups, then we will return nil
+func (r *Disruption) validateUserInfoGroup() error {
+	if len(permittedUserGroups) == 0 {
+		return nil
+	}
+
+	userInfo, err := r.UserInfo()
+	if err != nil {
+		return err
+	}
+
+	validGroupFound := false
+
+	for _, group := range userInfo.Groups {
+		_, ok := permittedUserGroups[group]
+		if ok {
+			validGroupFound = true
+
```

Comment:
Mostly useless improvement but I think we can break the loop once one permitted group has been found.